config/300-topic.yaml

```diff
@@ -0,0 +1,101 @@
+# Copyright 2019 The Knative Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+apiVersion: apiextensions.k8s.io/v1beta1
+kind: CustomResourceDefinition
+metadata:
+ name: topics.pubsub.cloud.run
+ labels:
+   events.cloud.run/release: devel
+   events.cloud.run/crd-install: "true"
+spec:
+  group: pubsub.cloud.run
+  version: v1alpha1
+  names:
+    kind: Topic
+    plural: topics
+    singular: topic
+    categories:
+    - all
+    - knative
+    - pubsub
+  scope: Namespaced
+  subresources:
+    status: {}
+  additionalPrinterColumns:
+    - name: Ready
+      type: string
+      JSONPath: ".status.conditions[?(@.type==\"Ready\")].status"
+    - name: Reason
+      type: string
+      JSONPath: ".status.conditions[?(@.type==\"Ready\")].reason"
+    - name: Hostname
+      type: string
+      JSONPath: .status.address.hostname
+    - name: Age
+      type: date
+      JSONPath: .metadata.creationTimestamp
+  validation:
+    openAPIV3Schema:
+      properties:
+        spec:
+          properties:
+            secret:
+              type: object
+              # TODO: document we default.
+              description: "Credential to use to manage Cloud Pub/Sub. The value of the secret entry must be a service account key in the JSON format (see https://cloud.google.com/iam/docs/creating-managing-service-account-keys)."
+            project:
+              type: string
+              # TODO: document we default.
+              description: "ID of the Google Cloud Project to own the Pub/Sub credentials. E.g. 'my-project-1234' rather than its display name, 'My Project' or its number '1234567890'."
+            topic:
+              type: string
+              description: "ID of the Cloud Pub/Sub Topic to create. It must be in the form of the unique identifier within the project, not the entire name. E.g. it must be 'laconia', not 'projects/my-gcp-project/topics/laconia'."
+            propagationPolicy:
+              type: string
```

Comment:
Doesn't OpenAPI schema support enums - for policy?

Reply:
hm, we have never used that feature. Followup.

Reply:
https://github.com/GoogleCloudPlatform/cloud-run-events/issues/38